src/docdb/tree/DocDBDocumentsTreeItem.ts

```diff
@@ -57,39 +58,33 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
 
         if (docID || docID === "") {
             docID = docID.trim();
-            let body = { id: docID };
-            body = <NewDocument>(await this.promptForPartitionKey(body));
+            let body: ItemDefinition = { id: docID };
+            body = (await this.promptForPartitionKey(body));
             context.showCreatingTreeItem(docID);
-            const document = await this.createDocument(body);
+            const item: ItemResponse<ItemDefinition> = await this.root.getDocumentClient().database(this.parent.parent.id).container(this.parent.id).items.create(body);
 
-            return this.initChild(document);
+            return this.initChild(nonNullProp(item, 'resource'));
         }
 
         throw new UserCancelledError();
     }
 
-    public async createDocument(body: NewDocument): Promise<RetrievedDocument> {
-        return await new Promise<RetrievedDocument>((resolve, reject) => {
-            this.root.getDocumentClient().createDocument(this.link, body, (err, result: RetrievedDocument) => {
-                if (err) {
-                    reject(err);
-                } else {
-                    resolve(result);
-                }
-            });
-        });
+    public async createDocument(body: ItemDefinition): Promise<ItemDefinition> {
+        const item: ItemResponse<ItemDefinition> = await this.root.getDocumentClient().database(this.parent.id).container(this.id).items.create(body);
+        return nonNullProp(item, 'resource');
     }
 
     public documentHasPartitionKey(doc: Object): boolean {
         let interim = doc;
-        let partitionKey: string | undefined = this.parent.partitionKey && this.parent.partitionKey.paths[0];
-        if (!partitionKey) {
+        const partitionKey = this.parent.partitionKey && this.parent.partitionKey;
```

Comment:
Please add the new type here. Also how come it has a "paths" object that's an array now? And are we sure we should only use the first path?